### PR TITLE
PHPCS: EU Cookie Widget

### LINKED ISSUE #21991: PHPCS: EU Cookie Widget
### Impacted plugin

Jetpack

### What

PHPCS

### How

```
	"projects/plugins/jetpack/modules/widgets/eu-cookie-law.php",
	"projects/plugins/jetpack/modules/widgets/eu-cookie-law/form.php",
	"projects/plugins/jetpack/modules/widgets/eu-cookie-law/widget.php",
```

### PR DESCRIPTION
**Pull request by samiff**
Fixes #21991

#### Changes proposed in this Pull Request:

PHPCS for EU Cookie Widget files.

#### Jetpack product discussion

#21991

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

Linting of the affected files should produce no errors/warnings:

```sh
composer phpcs:lint projects/plugins/jetpack/modules/widgets/class-jetpack-eu-cookie-law-widget.php projects/plugins/jetpack/modules/widgets/eu-cookie-law/form.php projects/plugins/jetpack/modules/widgets/eu-cookie-law/widget.php
```

With changes pulled down, and error logs & browser console being monitored, add the Cookie & Consents widget to your site and try editing _all the things_ to make sure nothing breaks.

### CODE REVIEW

**Comment by samiff** [comment]
Two items someone might chime in on when reviewing:

1. In `projects/plugins/jetpack/modules/widgets/class-jetpack-eu-cookie-law-widget.php` ignoring the `OutputNotEscaped` rule on `before_widget` and `after_widget`. Seems like this was what other widgets have used, but wasn't 100% on this.
2. In `projects/plugins/jetpack/modules/widgets/eu-cookie-law/form.php` this is an included file that references `$instance` and `$this` frequently in the file. Wasn't sure if disabling `UndefinedVariable` for the entire file was acceptable for something like this, or if there was a better way?

File: projects/plugins/jetpack/modules/widgets/class-jetpack-eu-cookie-law-widget.php [hunk 1]
```diff
@@ -222,20 +227,21 @@ public function form( $instance ) {
...
-			require( dirname( __FILE__ ) . '/eu-cookie-law/form.php' );
+			require __DIR__ . '/eu-cookie-law/form.php';
 		}
 
 		/**
 		 * Sanitize widget form values as they are saved.
 		 *
 		 * @param  array $new_instance Values just sent to be saved.
-		 * @param  array $old_instance Previously saved values from database.
+		 * @param array $old_instance Previously saved values from database.
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/widgets/class-jetpack-eu-cookie-law-widget.php:241]: Probably both lines should have either one space or two between `@param` and the type.

I'd do one space, but I don't really care which any more than phpcs does.

File: projects/plugins/jetpack/modules/widgets/eu-cookie-law/form.php [hunk 2]
```diff
@@ -1,7 +1,13 @@
 <?php
+/**
+ * EU Cookie Law Widget form.
+ *
+ * @package automattic/jetpack
+ */
 
 use Automattic\Jetpack\Redirect;
 
+// phpcs:disable VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/widgets/eu-cookie-law/form.php:10]: I suppose if you really wanted to you could do something like https://github.com/Automattic/jetpack/blob/6d1bce942323a8735acad9bffa4cefb626e298d4/projects/plugins/beta/src/admin/exception.template.php#L13-L17 to define the two variables that are supposed to be supplied by the outer scope.

But if you don't want to, that's ok with me.

**Review by anomiex — APPROVED** [review]
Two nitpicks inline, but nothing worth withholding approval for if you don't want to deal with them.

> 1. In `projects/plugins/jetpack/modules/widgets/class-jetpack-eu-cookie-law-widget.php` ignoring the `OutputNotEscaped` rule on `before_widget` and `after_widget`. Seems like this was what other widgets have used, but wasn't 100% on this.

That's fine IMO, for exactly that reason. Probably best to keep consistent with WP Core's widgets that don't escape those either.

File: projects/plugins/jetpack/modules/widgets/eu-cookie-law/form.php (hunk 2 above)
Comment (samiff) [reply projects/plugins/jetpack/modules/widgets/eu-cookie-law/form.php:10]: Thanks for sharing that example 👍  I think I'll leave as-is for now to better match other files already using the ignore.

**Review by anomiex — APPROVED** [review]

**Pull request merged**